test/dynamo/test_subclasses.py

```diff
@@ -3051,6 +3070,151 @@ def arg_fn(values=values, offsets=offsets):
 
         self._validate_compile(fn, arg_fn)
 
+    # See [ Best effort SymInt association ] for more info.
+    #
+    # This test is expected to fail because doing
+    # nested_tensor_from_jagged creates a fresh cache, and creates
+    # a new ephemeral symbol, which fails when it needs to be
+    # involved in any guard. In an ideal world, the resulting nt
+    # from nested_tensor_from_jagged preserves the original symbol
+    # which still has a proper source, but that is too hard.
+    @unittest.expectedFailure
+    @torch.fx.experimental._config.patch(use_duck_shape=False)
```

Comment:
not familiar with this one - what's it for?

Reply:
With duck sizing enabled, two inputs that share the same value are assumed to be the same. It achieves this by mapping val to symbol, and if a  newly created symints has the same val as a previous symint, the same symbol is reused.

If duck sizing were enabled, intermediate_nt may share the same symbol as nt (which does have a proper source because nt was passed in as input), and that would side step the issue I want to expose here which relies on intermediate_nt having a symbol with ephemeral source.